wfe/wfe.go

```diff
@@ -1728,14 +1687,12 @@ func (wfe *WebFrontEndImpl) revokeCertByKeyID(
 		return prob
 	}
 
-	keyID, err := keyToID(key)
+	existingAcct, err := wfe.db.GetAccountByKey(key)
 	if err != nil {
-		return acme.MalformedProblem(err.Error())
+		return acme.MalformedProblem("Cannot obtain key ID from public key")
 	}
-
-	existingAcct := wfe.db.GetAccountByID(keyID)
 	if existingAcct == nil {
-		return acme.UnauthorizedProblem(fmt.Sprintf("Account with keyID %q does not exist", keyID))
+		return acme.UnauthorizedProblem("Account with public key does not exist")
```

Comment:
for this error message, I think it might be clearer as: "No account found corresponding to public key authenticating this request", what do you think?

Reply:
Yes, that's much better. I'm using your text now.